src/Controls/src/Core/HandlerImpl/SwipeView/SwipeView.Impl.cs

```diff
@@ -112,6 +116,49 @@ private protected override void OnParentChangedCore()
 			}
 		}
 
+		internal void AddLogicalChild(Element element)
+		{
+			if (element == null)
+			{
+				return;
+			}
+
+			if (_logicalChildren.Contains(element))
+				return;
+
+			_logicalChildren.Add(element);
+			element.Parent = this;
+			OnChildAdded(element);
+			VisualDiagnostics.OnChildAdded(this, element);
+		}
+
+		internal void RemoveLogicalChild(Element element)
+		{
+			if (element == null)
+			{
+				return;
+			}
+
+			element.Parent = null;
+
+			if (!_logicalChildren.Contains(element))
+				return;
+
+			var oldLogicalIndex = _logicalChildren.IndexOf(element);
+			_logicalChildren.Remove(element);
+			OnChildRemoved(element, oldLogicalIndex);
+			VisualDiagnostics.OnChildRemoved(this, element, oldLogicalIndex);
```

Comment:
You are iterating over the list more times than needed, the RemoveLoicailChild is taking the element via the index, then you try to find the index again, that means you are recomputing data which you already had when you called the method. This can be refactored to be more efficient.

Reply:
Changed!